File: onderwijs4u-back-end/src/main/java/nl/han/devops/data/AuthenticationDAO.java
package nl.han.devops.data;

import nl.han.devops.repositories.UserRepository;
import org.springframework.stereotype.Component;

@Component
public class AuthenticationDAO {
    private final UserRepository userRepository;

    public AuthenticationDAO(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public boolean tokenExists(String token) {
        return userRepository.findByToken(token).isPresent();
    }
}

File: back-end/src/main/kotlin/nl/lpp/uncenshow/datasource/authentication/AuthenticationDAO.kt
package nl.lpp.uncenshow.datasource.authentication

import nl.lpp.uncenshow.dto.user.UserRepository
import org.springframework.stereotype.Component

@Component
class AuthenticationDAO (private val userRepository: UserRepository){

    fun tokenExists(token: String): Boolean {
        return userRepository.findItemByToken(token) != null
    }
}